card/mod/02_basic_types/set/type/pointer.rb

```diff
@@ -4,6 +4,11 @@
   self.drop_item Env.params['drop_item'] if Env.params['drop_item']
 end
 
+event :insert_item_event, :before=>:approve, :on=>:save, :when=> proc {|c| Env.params['insert_item']} do
+  index = Env.params['item_index'] || 0
+  self.insert_item index.to_i, Env.params['insert_item']
+end
+
```

Comment:
probably right that it's not needed.

in my experience, the most common time this is needed is when you're doing assigments.  Imagine, for example, that you've got an accessor like "age".  When you do:
````
def gonna_die_in
  death_age - age
end
````
...there's no need for self.  But if you do:
````
def have_birthday
  self.age = age + 1
end
````
...then you need that "self" to avoid creating a local variable "age"